evalml/tests/model_understanding_tests/test_graphs.py

```diff
@@ -1028,3 +1030,39 @@ class NotTSPipeline:
     error_msg = "graph_prediction_vs_actual_over_time only supports time series regression pipelines! Received regression."
     with pytest.raises(ValueError, match=error_msg):
         graph_prediction_vs_actual_over_time(NotTSPipeline(), None, None, None)
+
+
+def test_visualize_decision_trees(tree_estimators, logit_estimator, tmpdir):
+    graphviz = pytest.importorskip('graphviz', reason='Skipping visualizing test because graphviz not installed')
+    est_class, est_reg = tree_estimators
+    est_logit = logit_estimator
+
+    filepath = os.path.join(str(tmpdir), 'invalid', 'path', 'test.png')
+    assert not os.path.exists(filepath)
+    with pytest.raises(ValueError, match="Specified filepath is not writeable"):
+        visualize_decision_tree(clf=est_class, filepath=filepath)
+
+    filepath = os.path.join(str(tmpdir), 'test_0.xyz')
+    with pytest.raises(ValueError, match=f"Unknown format 'xyz'. Make sure your format is one of the following: {graphviz.backend.FORMATS}"):
+        visualize_decision_tree(clf=est_class, filepath=filepath)
+
+    filepath = os.path.join(str(tmpdir), 'test_1.png')
+    with pytest.raises(ValueError, match="Tree visualizations are not supported for non-Tree estimators"):
+        visualize_decision_tree(clf=est_logit, filepath=filepath)
+
+    with pytest.raises(ValueError, match="Unknown value: '-1'. The parameter max_depth has to be a non-negative integer"):
+        visualize_decision_tree(clf=est_class, max_depth=-1, filepath=filepath)
+
+    filepath = os.path.join(str(tmpdir), 'test_2')
+    src = visualize_decision_tree(clf=est_class, filled=True, max_depth=3, rotate=True, filepath=filepath)
+    assert src.format == 'pdf'  # Check that extension defaults to pdf
+    assert isinstance(src, graphviz.Source)
+
+    filepath = os.path.join(str(tmpdir), 'test_3.pdf')
+    src = visualize_decision_tree(clf=est_reg, filled=True, filepath=filepath)
+    assert src.format == 'pdf'
+    assert isinstance(src, graphviz.Source)
+
+    src = visualize_decision_tree(clf=est_reg, filled=True, max_depth=2)
+    assert src.format == 'pdf'
+    assert isinstance(src, graphviz.Source)
```

Comment:
Can you add a test to see what happens if we try `visualize_decision_tree` on a Decision Tree that hasn't yet been fit to data? As well as if we fit it twice (on different data), will the corresponding visualization/source change for each tree?